### DISCUSSION TITLE
Is there a way to apply lazy to json() data?

### DISCUSSION THREAD

**Discussion by eyedroot** [spatie/laravel-data#250] [Q&A] [answered]
I use the laravel-data library to create DTO objects from JSON data.

I want to convert a part of `json()` data received through `Http::get()->json()` to **Lazy**, but I am wondering if there is a way.

Below is an example code (doesn't work).

```php
return DataDto::from(Http::get('some-url')->json());
```

```php
class DataDto extends Data
{
  public function __construct(
    public readonly Lazy|ChildDto $data,
    public readonly VersionDto $version,
  ) {}

  public static function fromModel(array $json): self
  {
    return new self(
      Lazy::create(fn () => ChildDto::from($json['data']),
      VersionDto::from($data['version']),
    );
  }
}
```

Thanks for reading. 🙏

**Comment by rubenvanassche** [comment, marked as answer]
Hi @binbyz,

This code looks correct, what's the type and contents of `Http::get('some-url')->json()`? is the `fromModel` method triggered when you run this code (you can check this with a dd or ray).

**Reply by eyedroot** [reply]
@rubenvanassche thanks for replying I miss :)
